connectivity/manifests/client-egress-only-dns.yaml

```diff
@@ -22,3 +22,22 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: coredns
+  - toPorts:
+    - ports:
+      - port: "53"
+        protocol: ANY
+      rules:
+        dns:
+        - matchPattern: "*"
+    toCIDR:
+    - 169.254.0.0/16
+    - fd00::/8
+  - toPorts:
+    - ports:
+      - port: "53"
+        protocol: ANY
+      rules:
+        dns:
+        - matchPattern: "*"
+    toEntities:
+    - host
```

Comment:
I'm hesitant to attribute the host identity to 169.254 CIDR since it doesn't necessarily mean local host only traffic. For example, the subnet is used for a remote metadata server in EKS deployments.

Reply:
^ Maybe relevant, I wonder if you are able to describe why both rules are necessary - it looks like this policy allows either to the 169.254.0.0/16 CIDR on port 53, or separately also to the host on port 53. Did you find that something didn't work if one of these rules is not present?

Ideally we'd have some way to detect whether node-local DNS was in use and decide to dynamically insert these policies based on that. In a sense, this is what we have with the existing kube-dns / coredns label matching logic, since pods would have to first be deployed with those labels for the policy to apply. But I don't know if there's an easy way to do that.

While it's maybe not ideal to poke holes for node-local DNS if it's not configured, I don't see a huge danger here as the user already has access to deploy pods and configure the networking policy for these pods, so they can control what the pods have access to; and furthermore, it's a bit hard to see how an adversary could leverage these policies if this is limited to the connectivity checks and also if they would have to deploy some rogue DNS server on the host or somehow hijack 169.254.x.y traffic on the node first before they can leverage this policy --- which would imply that they already have bigger privileges than what these configurations would allow.

Reply:
> I wonder if you are able to describe why both rules are necessary - it looks like this policy allows either to the 169.254.0.0/16 CIDR on port 53, or separately also to the host on port 53. Did you find that something didn't work if one of these rules is not present?

@joestringer Yes, it is due to https://github.com/cilium/cilium/pull/18672. Depending on the order in which cilium pod & node-local-dns pod are started, either one or the other policy works. Only both policies work for all the cases.

Reply:
> I'm hesitant to attribute the host identity to 169.254 CIDR since it doesn't necessarily mean local host only traffic. For example, the subnet is used for a remote metadata server in EKS deployments.

@aditighag I agree that whole link-local subnet does not necessarily mean localhost-only traffic. But I wasn't able to figure out anything else that wouldn't be too complex - selecting the pod by labels does not work in this case (not sure if that is a bug or not).

The only other option that I see would be to have the node-local-dns IP address configurable or auto-detected somehow. In CLI case this is doable, but not sure how to easily implement that in the manifest-based deployment (see https://github.com/cilium/cilium/pull/18672).

After all, these are test, aren't they? So I believe the discussion should be about how/when this change could affect the test results. I cannot think about such scenarios - and if there are some, I think they should be definitely less common than the standard node-local-dns deployment.

Reply:
My gut feeling here is that we should figure out the solution to #18644 first, then based on that solution we will have a better understanding of how these connectivity tests need to be written to account for the (hopefully now more reliable) implementation.

Reply:
@joestringer OK, let me close this PR and wait for the resolution of #18644